apps/web/src/components/Publication/OpenAction/UnknownModule/Decent/CurrencySelector.tsx

```diff
@@ -0,0 +1,198 @@
+import type { FC } from 'react';
+import type { Address } from 'viem';
+
+import { STATIC_IMAGES_URL } from '@hey/data/constants';
+import getRedstonePrice from '@hey/helpers/getRedstonePrice';
+import getTokenImage from '@hey/helpers/getTokenImage';
+import { useQuery } from '@tanstack/react-query';
+import { useAllowedTokensStore } from 'src/store/persisted/useAllowedTokensStore';
+import { formatUnits } from 'viem';
+import { useAccount, useBalance } from 'wagmi';
+
+const OA_SUPPORTED_CURRENCIES = ['USDC', 'WETH', 'WMATIC'];
+
+interface CurrencySelectorProps {
+  onSelectCurrency: (currency: Address) => void;
+}
+
+const CurrencySelector: FC<CurrencySelectorProps> = ({ onSelectCurrency }) => {
+  const { allowedTokens } = useAllowedTokensStore();
+  const { address } = useAccount();
+
+  const { data: wmaticBalanceData, isLoading: wmaticBalanceLoading } =
+    useBalance({
+      address,
+      chainId: 137,
+      query: { refetchInterval: 10000 },
+      token: allowedTokens.find((token) => token.symbol === 'WMATIC')
+        ?.contractAddress as Address
```

Comment:
we can create a method called getToken(symbol)? also i'd say compare with token address instead of symbol which is more unique, symbol might be same on future cases

Reply:
refer to the changes done by @defispartan

> Re-wrote this component to use useRatesStore for USD prices, add new helper function to format token and usd amounts, and query tokens based on addresses instead of symbol